### PR TITLE
Fix smarty condition count

### PR DESCRIPTION
**Pull request by tomlev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | fix weird bug on some smarty/php versions
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | add a cart rule

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9170)

### CODE REVIEW

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Wait for ci before merge

**Pull request merged**
